src/vector/platform/WebPlatform.ts

```diff
@@ -46,7 +46,36 @@ export default class WebPlatform extends VectorBasePlatform {
         super();
         // Register service worker if available on this platform
         if ("serviceWorker" in navigator) {
-            navigator.serviceWorker.register("sw.js");
+            // sw.js is exported by webpack, sourced from `/src/serviceworker/index.ts`
+            const swPromise = navigator.serviceWorker.register("sw.js");
+
+            // Jest causes `register()` to return undefined, so swallow that case.
+            if (swPromise) {
+                swPromise
+                    .then(async (r) => {
+                        // always ask the browser to update. The browser might not actually do it, but at least we asked.
+                        await r.update();
+                        return r;
+                    })
+                    .then((r) => {
+                        navigator.serviceWorker.addEventListener("message", (e) => {
+                            try {
+                                if (e.data?.["type"] === "userinfo" && e.data?.["responseKey"]) {
+                                    const userId = localStorage.getItem("mx_user_id");
+                                    const deviceId = localStorage.getItem("mx_device_id");
+                                    r.active!.postMessage({
+                                        responseKey: e.data["responseKey"],
+                                        userId,
+                                        deviceId,
+                                    });
+                                }
+                            } catch (e) {
+                                console.error("Error responding to service worker: ", e);
+                            }
+                        });
+                    })
+                    .catch((e) => console.error("Error registering/updating service worker:", e));
+            }
```

Comment:
`then` and `catch` are so last-decade. Can we factor this out to a separate `async function registerServiceWorker`, and just do

```typescript
    registerServiceWorker().catch((e) => console.error("Error registering/updating service worker:", e));
```